### ISSUE TITLE
Improved Project Create Experience

### ISSUE THREAD

**Issue by ntotten** [forcedotcom/salesforcedx-vscode#705]
## Is your feature request related to a problem?
> Note: This feature is mostly related to the SFDX CLI, but I am putting it here for now as the CLI doesn't (yet) have a public repo.

Today, you have basically two options for creating a project: a regular project and a project with a manifest. However, there are many different options that could be added to the default project experience.

**Describe the solution you'd like**
When creating a project using the loopback CLI tool you are given a number of different options as shown below.

![image](https://user-images.githubusercontent.com/282782/47649669-597e8600-db54-11e8-8a33-e2862a0c4bc4.png)

The SFDX CLI should allow for similar extensibility and flexibiity when creating new project.

## Project Template Options
The following options could be added to the new project create experience today and looks somethng like the following.

```bash
sfdx force:project:create
? Project name: <name>
? Default package directory: <name|force-app>
? API version: <versionNumber|current> # Probably advanced only
? Logint Url: <loginUrl|https://login.salesforce.com> # Probably advanced only
? Select features to enable in the project:
  [ ] Include default manifest
  [ ] Enable prettier
  [ ] Enable jslint
  [ ] Enable PMD
  [ ] Enable Visual Studio Code
  [ ] Enable Git
```

The options are as follows:
1. **Include default manifest**: this is the same thing that the `--manifest` option does today.
1. **Enabled prettier**: Adds a `package.json` file (if needed) with prettier installed (and eventually with the prettier-apex plugin). Includes a script in `package.json` for `prettier` to format all code in the reposititory
1. **Enable jslint**: Adds a `package.json` file (if needed) with jslint installed. Includes a script for `lint` that runs jslint. Also includes a recommended config for jslint.
1. **Enable PMD**: Adds a recommended PMD ruleset to the project and a `package.json` script to run PMD.
1. **Enable Visual Studio Code**: Adds a `.vscode` directory to the project with recommended settings in `settings.json` as well as the default configurations for the replay debugger in `launch.json`
1. **Enable Git**: Initializes the directory as a git repository and adds the recommended `.gitignore` file to the project.

## Implimenation in VS Code
In VS Code the options would be presented as dialogs that either prompt for text values (in the case of things like the `Project name`) or yes/no options in the case of the features. There would be a single command `SFDX: Create Project` instead of multiple commands today like `SFDX: Create Project with Manifest`

**Comment by vazexqi** [comment]
Just FYI for the VS code side, we need to think about the UX. We did consider the approach of https://github.com/SamVerschueren/vscode-yo (see the gif at the site), but, at that time, decided that the lack of proper menus means that the experience was actually worse and more confusing that the current approach.

**Comment by ntotten** [comment]
I don’t think that experience is too bad, but yeah it certainly could be better. I think for new users though it’s helpful.

The other thing is we could have a standard VSCode default. So maybe we have “SFDX: Create project” and “SFDX: Create project with custom options” or something.

**Comment by ntotten** [comment]
//cc @claire-bianchi

**Comment by ntotten** [comment]
I hadn't seen this before. The ability to indicate which step you are on in a multi-step dialog: https://code.visualstudio.com/updates/v1_26#_quickinput-api
